File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.py
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        prefixSum = [0]
        for num in arr:
            prefixSum.append(prefixSum[-1] + num)
        ans = 0
        for i in range(len(arr)):
            for j in range(i,len(arr),2):
                
                ans += prefixSum[j+1] - prefixSum[i]
        return ans
        

File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        
        int ans,n;
        ans = 0;
        n = arr.length;
        
        for (int i = 0; i< n;i++){
            int count = ((i+1) * (n-i) +1) / 2;
            ans += count * arr[i];
    
        }
        
        return ans;
        
    }
}